src/components/vm/nics/vmNicsCard.jsx

```diff
@@ -389,7 +389,11 @@ export class VmNetworkTab extends React.Component {
         detailMap = detailMap.filter(d => !d.hidden);
 
         const columnTitles = detailMap.map(target => target.name);
-        const rows = vm.interfaces.sort().map(target => {
+        // Normally we should identify a vNIC to detach by a number of slot, bus, function and domain.
+        // Such detachment is however broken in virt-xml, so instead let's detach it by the index of <interface> in array of VM's XML <devices>
+        // This serves as workaround for https://github.com/virt-manager/virt-manager/issues/356
+        const ifaces = vm.interfaces.map((iface, index) => ({ ...iface, index }));
+        const rows = ifaces.sort().map(target => {
```

Comment:
`ifaces` is a list of objects with no obvious search key, so it will be sorted by some arbitrary object hash or pointer or so. Shouldn't you give a comparison function to pick out the interface name, or sort by connectionName, then by VM name (to group related interfaces together), then by name or so?

This isn't new in this PR, but let's fix this.

Reply:
Yeah this sort() there doesn't make sense. I added a custom sort function which first tries to sort interfaces by type and then by MAC.

Reply:
Nice, thanks!